### PR TITLE
Added MOF descriptions to all resources

### PR DESCRIPTION
**Pull request by BrianFarnhill**
@camiloborges @anlynes @caadam @ykuijs Tagging for a code review guys, help a brother out!

### CODE REVIEW

File: Modules/xSharePoint/DSCResources/MSFT_xSPInstall/MSFT_xSPInstall.schema.mof [hunk 1]
```diff
@@ -31,8 +31,8 @@ Currently SharePoint Foundation is not supported by xSPInstall (see [Issue #81](
 [ClassVersion("1.0.0.0"), FriendlyName("xSPInstall")]
 class MSFT_xSPInstall : OMI_BaseResource
 {
-    [Key] String BinaryDir;
-    [Required] String ProductKey;
-    [Required, ValueMap{"Present","Absent"}, Values{"Present","Absent"}] string Ensure;
+    [Key, Description('The direcotry that contains all of the SharePoint binaries')] String BinaryDir;
```
Comment (ykuijs) [inline Modules/xSharePoint/DSCResources/MSFT_xSPInstall/MSFT_xSPInstall.schema.mof]: Typo: directory

File: Modules/xSharePoint/DSCResources/MSFT_xSPManagedAccount/MSFT_xSPManagedAccount.schema.mof [hunk 2]
```diff
@@ -17,10 +17,10 @@ The settings for EmailNotification, PreExpireDays and Schedule all relate to ena
...
-    [Key] string AccountName;
-    [Required, EmbeddedInstance("MSFT_Credential")] String Account;
-    [Write, EmbeddedInstance("MSFT_Credential")] String InstallAccount;
-    [Write] Uint32 EmailNotification;
-    [Write] Uint32 PreExpireDays;
-    [Write] string Schedule;
+    [Key, Description('The username of the account')] string AccountName;
+    [Required, Description('The credential with password of the account'), EmbeddedInstance("MSFT_Credential")] String Account;
+    [Write, Description('How many days before a password change should an email be sent')] Uint32 EmailNotification;
+    [Write, Description('How many days before a passowrd expires should it be changed')] Uint32 PreExpireDays;
```
Comment (ykuijs) [inline Modules/xSharePoint/DSCResources/MSFT_xSPManagedAccount/MSFT_xSPManagedAccount.schema.mof]: Typo: password

Comment (ykuijs) [inline Modules/xSharePoint/DSCResources/MSFT_xSPManagedAccount/MSFT_xSPManagedAccount.schema.mof]: Typo: password

File: Modules/xSharePoint/DSCResources/MSFT_xSPCacheAccounts/MSFT_xSPCacheAccounts.schema.mof [hunk 3]
```diff
@@ -16,9 +16,9 @@ This resource is used to set the "super user" and "super reader" cache accounts
...
 {
-    [Key] string WebAppUrl;
-    [Required] string SuperUserAlias;
-    [Required] string SuperReaderAlias;
-    [Write, EmbeddedInstance("MSFT_Credential")] String InstallAccount;
+    [Key, Description('The URL of the web application to set the accounts for')] string WebAppUrl;
+    [Required, Description('The account name for the super user')] string SuperUserAlias;
+    [Required, Description('The account name fo the super reader')] string SuperReaderAlias;
+    [Write, Description('POWERSHELL 4 ONLY: The account to run this resource as, use PsDscRunAsAccount if using PowerShell 5'), EmbeddedInstance("MSFT_Credential")] String InstallAccount;
 };
```
Comment (ykuijs) [inline Modules/xSharePoint/DSCResources/MSFT_xSPCacheAccounts/MSFT_xSPCacheAccounts.schema.mof]: Off topic: Shouldn't this resource have a Ensure parameter as well?

File: Modules/xSharePoint/DSCResources/MSFT_xSPAppManagementServiceApp/MSFT_xSPAppManagementServiceApp.schema.mof [hunk 4]
```diff
@@ -19,10 +19,10 @@ Database names or server name will not be changed if the configuration does not
...
-    [Required] String ApplicationPool;
-    [Write] string DatabaseName;
-    [Write] String DatabaseServer;
-    [Write, EmbeddedInstance("MSFT_Credential")] String InstallAccount;
+    [Key, Description('The name of the app management service application')] string Name;
+    [Required, Description('The app pool that should be used to run the service app')] String ApplicationPool;
+    [Write, Description('The name of the database for the service application')] string DatabaseName;
+    [Write, Description('The name of the server for the database')] String DatabaseServer;
+    [Write, Description('POWERSHELL 4 ONLY: The account to run this resource as, use PsDscRunAsAccount if using PowerShell 5'), EmbeddedInstance("MSFT_Credential")] String InstallAccount;
 };
```
Comment (ykuijs) [inline Modules/xSharePoint/DSCResources/MSFT_xSPAppManagementServiceApp/MSFT_xSPAppManagementServiceApp.schema.mof]: Off topic: Shouldn't this resource have a Ensure parameter as well?

File: Modules/xSharePoint/DSCResources/MSFT_xSPQuotaTemplate/MSFT_xSPQuotaTemplate.schema.mof [hunk 5]
```diff
@@ -20,11 +20,11 @@ These settings will be used to make sure a certain quota template exists or not.
...
-    [Write] uint32 MaximumUsagePointsSolutions;
-    [Write] uint32 WarningUsagePointsSolutions;
-    [Required, ValueMap{"Present","Absent"}, Values{"Present","Absent"}] string Ensure;
-    [Write, EmbeddedInstance("MSFT_Credential")] String InstallAccount;
+    [Key, Description('The name of the quota template')] string Name;
+    [Write, Description('The maximum storage for sites of this template in MB')] uint32 StorageMaxInMB;
+    [Write, Description('The amount of storage for sites of this template that triggers a warning')] uint32 StorageWarningInMB;
+    [Write, Description('The maximum number of performance points for sandbox solutions for this template')] uint32 MaximumUsagePointsSolutions;
+    [Write, Description('The warning number of performance points for sandbox solutions for this template')] uint32 WarningUsagePointsSolutions;
+    [Required, Description('Present to create this tempalte, absent to ensure it does not exist'), ValueMap{"Present","Absent"}, Values{"Present","Absent"}] string Ensure;
```
Comment (ykuijs) [inline Modules/xSharePoint/DSCResources/MSFT_xSPQuotaTemplate/MSFT_xSPQuotaTemplate.schema.mof]: Typo: template

File: Modules/xSharePoint/DSCResources/MSFT_xSPSearchServiceApp/MSFT_xSPSearchServiceApp.schema.mof [hunk 6]
```diff
@@ -19,10 +19,10 @@ The database name parameter is used as the prefix for all search databases (so y
...
-    [Required] string ApplicationPool;
-    [Write] string DatabaseName;
-    [Write] string DatabaseServer;
-    [Write, EmbeddedInstance("MSFT_Credential")] String InstallAccount;
+    [Key, Description('The name of the search service application')] string Name;
+    [Required, Description('The application pool that it should run in')] string ApplicationPool;
+    [Write, Description('The name of the database (noting that some search databases will use this as a prefix)')] string DatabaseName;
+    [Write, Description('The server that host the databases for this service application')] string DatabaseServer;
+    [Write, Description('POWERSHELL 4 ONLY: The account to run this resource as, use PsDscRunAsAccount if using PowerShell 5'), EmbeddedInstance("MSFT_Credential")] String InstallAccount;
 };
```
Comment (ykuijs) [inline Modules/xSharePoint/DSCResources/MSFT_xSPSearchServiceApp/MSFT_xSPSearchServiceApp.schema.mof]: Off topic: Shouldn't this resource have a Ensure parameter as well?

File: Modules/xSharePoint/DSCResources/MSFT_xSPSecureStoreServiceApp/MSFT_xSPSecureStoreServiceApp.schema.mof [hunk 7]
```diff
@@ -19,17 +19,17 @@ The parameters passed in (except those related to database specifics) are valida
...
-    [Write] uint32 AuditlogMaxSize;
-    [Write, EmbeddedInstance("MSFT_Credential")] String DatabaseCredentials;
-    [Write] string DatabaseName;
-    [Write] string DatabaseServer;
-    [Write, ValueMap{"Windows","SQL"}, Values{"Windows","SQL"}] string DatabaseAuthenticationType;
-    [Write] string FailoverDatabaseServer;
-    [Write] boolean PartitionMode;
-    [Write] boolean Sharing;
-    [Write, EmbeddedInstance("MSFT_Credential")] String InstallAccount;
+    [Key, Description('The name of the sercure store service app')] string Name;
```
Comment (ykuijs) [inline Modules/xSharePoint/DSCResources/MSFT_xSPSecureStoreServiceApp/MSFT_xSPSecureStoreServiceApp.schema.mof]: Typo: secure

File: Modules/xSharePoint/DSCResources/MSFT_xSPSecureStoreServiceApp/MSFT_xSPSecureStoreServiceApp.schema.mof [hunk 8]
```diff
@@ -19,17 +19,17 @@ The parameters passed in (except those related to database specifics) are valida
...
+    [Write, Description('What is the maximum size of the audit log in MB')] uint32 AuditlogMaxSize;
+    [Write, Description('What SQL credentials should be used to access the database'), EmbeddedInstance("MSFT_Credential")] String DatabaseCredentials;
+    [Write, Description('The name of the database for the service app')] string DatabaseName;
+    [Write, Description('The name of the database server to host the database')] string DatabaseServer;
+    [Write, Description('What type of authentication should be used to access the database'), ValueMap{"Windows","SQL"}, Values{"Windows","SQL"}] string DatabaseAuthenticationType;
+    [Write, Description('The name of the database server hosting a failover instance of the database')] string FailoverDatabaseServer;
+    [Write, Description('Is partition mode enabled for this service app')] boolean PartitionMode;
+    [Write, Description('Is sharing enabled for this service app')] boolean Sharing;
+    [Write, Description('POWERSHELL 4 ONLY: The account to run this resource as, use PsDscRunAsAccount if using PowerShell 5'), EmbeddedInstance("MSFT_Credential")] String InstallAccount;
 };
```
Comment (ykuijs) [inline Modules/xSharePoint/DSCResources/MSFT_xSPSecureStoreServiceApp/MSFT_xSPSecureStoreServiceApp.schema.mof]: Off topic: Shouldn't this resource have a Ensure parameter as well?

File: Modules/xSharePoint/DSCResources/MSFT_xSPSessionStateService/MSFT_xSPSessionStateService.schema.mof [hunk 9]
```diff
@@ -18,10 +18,10 @@ If session timeout is not provided it will default to 60.
...
-    [Key] string DatabaseServer;
-    [Required, Write] boolean Enabled;
-    [Write] uint32 SessionTimeout;
-    [Write, EmbeddedInstance("MSFT_Credential")] String InstallAccount;
+    [Key, Description('The name of the database for the service')] string DatabaseName;
+    [Key, Description('The name of the database server for the database')] string DatabaseServer;
+    [Required, Description('Is the state service enabled')] boolean Enabled;
+    [Write, Description('What is the timeout on sessions')] uint32 SessionTimeout;
+    [Write, Description('POWERSHELL 4 ONLY: The account to run this resource as, use PsDscRunAsAccount if using PowerShell 5'), EmbeddedInstance("MSFT_Credential")] String InstallAccount;
 };
```
Comment (ykuijs) [inline Modules/xSharePoint/DSCResources/MSFT_xSPSessionStateService/MSFT_xSPSessionStateService.schema.mof]: Off topic: Shouldn't this resource have a Ensure parameter as well?

File: Modules/xSharePoint/DSCResources/MSFT_xSPSite/MSFT_xSPSite.schema.mof [hunk 10]
```diff
@@ -21,19 +21,19 @@ The current version of xSharePoint is only able to check for the existence of a
...
-    [Write] string Name;
-    [Write] string OwnerEmail;
-    [Write] string QuotaTemplate;
-    [Write] string SecondaryEmail;
-    [Write] string SecondaryOwnerAlias;
-    [Write] string Template;
-    [Write, EmbeddedInstance("MSFT_Credential")] String InstallAccount;
+    [Key, Description('The URL of the site collection')] string Url;
+    [Required, Description('The username of the site collection administrator')] string OwnerAlias;
+    [Write, Description('The compatability level of the site')] uint32 CompatibilityLevel;
```
Comment (ykuijs) [inline Modules/xSharePoint/DSCResources/MSFT_xSPSite/MSFT_xSPSite.schema.mof]: Typo: compatibility

File: Modules/xSharePoint/DSCResources/MSFT_xSPSite/MSFT_xSPSite.schema.mof [hunk 11]
```diff
@@ -21,19 +21,19 @@ The current version of xSharePoint is only able to check for the existence of a
...
+    [Write, Description('The name of the content database to create the site in')] string ContentDatabase;
+    [Write, Description('The description to apply to the site collection')] string Description;
+    [Write, Description('The URL of the host header web application to crete this site in')] string HostHeaderWebApplication;
```
Comment (ykuijs) [inline Modules/xSharePoint/DSCResources/MSFT_xSPSite/MSFT_xSPSite.schema.mof]: Typo: create

File: Modules/xSharePoint/DSCResources/MSFT_xSPStateServiceApp/MSFT_xSPStateServiceApp.schema.mof [hunk 12]
```diff
@@ -16,9 +16,9 @@ The database specific parameters are only used during initial provisioning of th
...
-    [Write, EmbeddedInstance("MSFT_Credential")] String DatabaseCredentials;
-    [Required] string DatabaseName;
-    [Write] string DatabaseServer;
-    [Write, EmbeddedInstance("MSFT_Credential")] String InstallAccount;
+    [Key, Description('The name of the state service app')] string Name;
+    [Write, Description('The database credentials for accessing the database'), EmbeddedInstance("MSFT_Credential")] String DatabaseCredentials;
+    [Required, Description('The name of the database for the service app')] string DatabaseName;
+    [Write, Description('The name of the database server')] string DatabaseServer;
+    [Write, Description('POWERSHELL 4 ONLY: The account to run this resource as, use PsDscRunAsAccount if using PowerShell 5'), EmbeddedInstance("MSFT_Credential")] String InstallAccount;
 };
```
Comment (ykuijs) [inline Modules/xSharePoint/DSCResources/MSFT_xSPStateServiceApp/MSFT_xSPStateServiceApp.schema.mof]: Off topic: Shouldn't this resource have a Ensure parameter as well?

File: Modules/xSharePoint/DSCResources/MSFT_xSPSubscriptionSettingsServiceApp/MSFT_xSPSubscriptionSettingsServiceApp.schema.mof [hunk 13]
```diff
@@ -19,10 +19,10 @@ Database names or server name will not be changed if the configuration does not
...
-    [Required] String ApplicationPool;
-    [Write] string DatabaseName;
-    [Write] String DatabaseServer;
-    [Write, EmbeddedInstance("MSFT_Credential")] String InstallAccount;
+    [Key, Description('The name of the subscription settings service app')] string Name;
+    [Required, Description('The name of the application pool the service app runs in')] String ApplicationPool;
+    [Write, Description('The name of the database for the service app')] string DatabaseName;
+    [Write, Description('The name of the database server')] String DatabaseServer;
+    [Write, Description('POWERSHELL 4 ONLY: The account to run this resource as, use PsDscRunAsAccount if using PowerShell 5'), EmbeddedInstance("MSFT_Credential")] String InstallAccount;
 };
```
Comment (ykuijs) [inline Modules/xSharePoint/DSCResources/MSFT_xSPSubscriptionSettingsServiceApp/MSFT_xSPSubscriptionSettingsServiceApp.schema.mof]: Off topic: Shouldn't this resource have a Ensure parameter as well?

File: Modules/xSharePoint/DSCResources/MSFT_xSPUsageApplication/MSFT_xSPUsageApplication.schema.mof [hunk 14]
```diff
@@ -19,14 +19,14 @@ The database settings are only used for initial provisioning, but the usage sett
...
+    [Write, Description('The name of the database for the service app')] string DatabaseName;
+    [Write, Description('The name of the database server')] string DatabaseServer;
+    [Write, Description('The credentials to use to access the database'), EmbeddedInstance("MSFT_Credential")] String DatabaseCredentials;
+    [Write, Description('The name of the failover database server')] string FailoverDatabaseServer;
+    [Write, Description('The time in minutes to cut over to new log files')] uint32 UsageLogCutTime;
+    [Write, Description('The location on each server to store the log files')] string UsageLogLocation;
+    [Write, Description('The maximum file size for log files in KB')] uint32 UsageLogMaxFileSizeKB;
+    [Write, Description('The total space of all log files on disk in GB')] uint32 UsageLogMaxSpaceGB;
+    [Write, Description('POWERSHELL 4 ONLY: The account to run this resource as, use PsDscRunAsAccount if using PowerShell 5'), EmbeddedInstance("MSFT_Credential")] String InstallAccount;
 };
```
Comment (ykuijs) [inline Modules/xSharePoint/DSCResources/MSFT_xSPUsageApplication/MSFT_xSPUsageApplication.schema.mof]: Off topic: Shouldn't this resource have a Ensure parameter as well?

File: Modules/xSharePoint/DSCResources/MSFT_xSPUserProfileSyncConnection/MSFT_xSPUserProfileSyncConnection.schema.mof [hunk 15]
```diff
@@ -23,16 +23,15 @@ This resource currently supports AD only.
...
-    [write] string Server;
-    [Write] boolean UseSSL;
-    [Write] boolean Force;
-    [Write, ValueMap{"ActiveDirectory","BusinessDataCatalog"}, Values{"ActiveDirectory","BusinessDataCatalog"}] string ConnectionType ;
+    [Key, Description('The name of the connection')] string Name;
+    [Required, Description('The name of the AD forest to read from')] string Forest;
+    [Required, Description('The name of the user profile service that this connection is attached to')] string UserProfileService;
+    [Required, Description('The credentials to connect to Active Directory with'), EmbeddedInstance("MSFT_Credential")] string ConnectionCredentials;    
+    [Required, Description('A listo f the OUs to import users from')] string IncludedOUs[];
+    [Write, Description('A list of the OUs to ignroe users from')] string ExcludedOUs[];
```
Comment (ykuijs) [inline Modules/xSharePoint/DSCResources/MSFT_xSPUserProfileSyncConnection/MSFT_xSPUserProfileSyncConnection.schema.mof]: Typo: ignore

File: Modules/xSharePoint/DSCResources/MSFT_xSPVisioServiceApp/MSFT_xSPVisioServiceApp.schema.mof [hunk 16]
```diff
@@ -13,6 +13,6 @@ The resource will provision and configure the Visio Graphics Service Application
 [ClassVersion("1.0.0.0"), FriendlyName("xSPVisioServiceApp")]
 class MSFT_xSPVisioServiceApp : OMI_BaseResource
 {
-    [Key] string Name;
-    [Required] string ApplicationPool;
+    [Key, Description('The name of the service application')] string Name;
+    [Required, Description('The name of the application pool to run the service app in')] string ApplicationPool;
 };
```
Comment (ykuijs) [inline Modules/xSharePoint/DSCResources/MSFT_xSPVisioServiceApp/MSFT_xSPVisioServiceApp.schema.mof]: Off topic: Shouldn't this resource have a Ensure parameter as well?

File: Modules/xSharePoint/DSCResources/MSFT_xSPWebAppGeneralSettings/MSFT_xSPWebAppGeneralSettings.schema.mof [hunk 17]
```diff
@@ -19,21 +19,21 @@ Any settings not included will be left as the default (or whatever they have bee
...
-    [write] boolean RecycleBinCleanupEnabled;
-    [write] uint32 RecycleBinRetentionPeriod;
-    [write] uint32 SecondStageRecycleBinQuota;
-    [write] uint32 MaximumUploadSize;
-    [write] boolean CustomerExperienceProgram;
-    [write] boolean PresenceEnabled;
-    [Write, EmbeddedInstance("MSFT_Credential")] string InstallAccount;
+    [Key, Description('The URL of the web app to set the general settings for')] string Url;
+    [Write, Description('The timezone code to use for this web app. A full list is at https://msdn.microsoft.com/en-us/library/office/microsoft.sharepoint.spregionalsettings.timezones.aspx')] uint32 TimeZone;
+    [Write, Description('Should alertd be enabled for this web app')] boolean Alerts;
```
Comment (ykuijs) [inline Modules/xSharePoint/DSCResources/MSFT_xSPWebAppGeneralSettings/MSFT_xSPWebAppGeneralSettings.schema.mof]: Typo: alert

File: Modules/xSharePoint/DSCResources/MSFT_xSPWebAppGeneralSettings/MSFT_xSPWebAppGeneralSettings.schema.mof [hunk 18]
```diff
@@ -19,21 +19,21 @@ Any settings not included will be left as the default (or whatever they have bee
...
+    [Write, Description('What is the maximum number of alerts that a user can create in this web app')] uint32 AlertsLimit;
+    [Write, Description('Should RSS feeds be enabled in this web app')] boolean RSS;
+    [Write, Description('Should the Blog API be enabled in this web app')] boolean BlogAPI;
+    [Write, Description('Is authentication required for the blog API')] boolean BlogAPIAuthenticated;
+    [Write, Description('What file handling mode should be used in this web app - strict or permissive'), ValueMap{"Strict","Permissive"}, Values{"Stric","Permissive"}] String BrowserFileHandling;
+    [Write, Description('Is security validation enforced in this web app')] boolean SecurityValidation;
+    [Write, Description('Is the recycle bin enabled in this web application')] boolean RecycleBinEnabled;
+    [Write, Description('Is automatic cleanup of the recycle bin enable in this web app')] boolean RecycleBinCleanupEnabled;
```
Comment (ykuijs) [inline Modules/xSharePoint/DSCResources/MSFT_xSPWebAppGeneralSettings/MSFT_xSPWebAppGeneralSettings.schema.mof]: Typo: enabled

File: Modules/xSharePoint/DSCResources/MSFT_xSPWebAppSiteUseAndDeletion/MSFT_xSPWebAppSiteUseAndDeletion.schema.mof [hunk 19]
```diff
@@ -20,10 +20,10 @@ You can enable or disable the Site Use and Deletion feature, specify the amount
 [ClassVersion("1.0.0"), FriendlyName("xSPWebAppSiteUseAndDeletion")]
 Class MSFT_xSPWebAppSiteUseAndDeletion : OMI_BaseResource
 {
-    [Key] string Url;
-    [write] boolean SendUnusedSiteCollectionNotifications;
-    [write] uint32 UnusedSiteNotificationPeriod;
-    [write] boolean AutomaticallyDeleteUnusedSiteCollections;
-    [write] uint32 UnusedSiteNotificationsBeforeDeletion;
-    [Write, EmbeddedInstance("MSFT_Credential")] string InstallAccount;
+    [Key, Description('the URL of the web application')] string Url;
```
Comment (ykuijs) [inline Modules/xSharePoint/DSCResources/MSFT_xSPWebAppSiteUseAndDeletion/MSFT_xSPWebAppSiteUseAndDeletion.schema.mof]: "The" should be starting with a capital

File: Modules/xSharePoint/DSCResources/MSFT_xSPWebApplication/MSFT_xSPWebApplication.schema.mof [hunk 20]
```diff
@@ -23,17 +23,17 @@ The resource will provision the web application with all of the current settings
...
-    [Write] string Path;
-    [Write] string Port;
-    [Write, EmbeddedInstance("MSFT_Credential")] string InstallAccount;
+    [Key, Description('The name of the web application')] string Name;
+    [Required, Description('The name of the application pool to run this site in')] string ApplicationPool;
+    [Required, Description('The name of the managed account to run the app pool with')] string ApplicationPoolAccount;
+    [Required, Description('The URL of the web application')] string Url;
+    [Write, Description('Should anonymous access be enabled for this web app')] boolean AllowAnonymous;
+    [Write, Description('What authentication mode should be used for the web app'), ValueMap{"NTLM","Kerberos"}, Values{"NTLM","Kerberos"}] string AuthenticationMethod;
+    [Write, Description('The name of the default content database for the web app')] string DatabaseName;
```
Comment (ykuijs) [inline Modules/xSharePoint/DSCResources/MSFT_xSPWebApplication/MSFT_xSPWebApplication.schema.mof]: Why "default content database"

File: Modules/xSharePoint/DSCResources/MSFT_xSPWorkManagementServiceApp/MSFT_xSPWorkManagementServiceApp.schema.mof [hunk 21]
```diff
@@ -23,14 +23,14 @@ Remarks
...
-    [Write] uint32 MinimumTimeBetweenEwsSyncSubscriptionSearches;
-    [Write] uint32 MinimumTimeBetweenProviderRefreshes;
-    [Write] uint32 MinimumTimeBetweenSearchQueries;
-    [Write] uint32 NumberOfSubscriptionSyncsPerEwsSyncRun;
-    [Write] uint32 NumberOfUsersEwsSyncWillProcessAtOnce;
-    [Write] uint32 NumberOfUsersPerEwsSyncBatch;
+    [Key, Description('The name of the work management service application')] string Name;
+    [write, Description('Present to ensure the app exists, absent to ensure it is removed'), ValueMap{"Present","Absent"}, Values{"Present","Absent"}] string Ensure;
+    [write, Description('The name of the application pool this will run in')] String ApplicationPool;
+    [Write, Description('The minimum amount of time bween EWA sync subscription searches')] uint32 MinimumTimeBetweenEwsSyncSubscriptionSearches;
```
Comment (ykuijs) [inline Modules/xSharePoint/DSCResources/MSFT_xSPWorkManagementServiceApp/MSFT_xSPWorkManagementServiceApp.schema.mof]: Typo: EWS

File: Modules/xSharePoint/DSCResources/MSFT_xSPWorkManagementServiceApp/MSFT_xSPWorkManagementServiceApp.schema.mof [hunk 22]
```diff
@@ -23,14 +23,14 @@ Remarks
...
+    [Write, Description('The minimum time between provider refreshes')] uint32 MinimumTimeBetweenProviderRefreshes;
+    [Write, Description('The minimum time between search queries')] uint32 MinimumTimeBetweenSearchQueries;
+    [Write, Description('The number of sunscription syncronisations per EWS sync run')] uint32 NumberOfSubscriptionSyncsPerEwsSyncRun;
```
Comment (ykuijs) [inline Modules/xSharePoint/DSCResources/MSFT_xSPWorkManagementServiceApp/MSFT_xSPWorkManagementServiceApp.schema.mof]: Typo: subscription

**Comment by ykuijs** [comment]
Added some comments for typo's and some off topic questions about missing Ensure parameters for several Service Applications. Probably not the correct place to correct them, but did want to mention them :-)

File: Modules/xSharePoint/DSCResources/MSFT_xSPManagedMetadataServiceApp/MSFT_xSPManagedMetaDataServiceApp.schema.mof [hunk 23]
```diff
@@ -19,9 +19,9 @@ The database server and database name properties are only used during provisioni
...
-    [Write, EmbeddedInstance("MSFT_Credential")] String InstallAccount;
-    [Required] string ApplicationPool;
-    [Write] string DatabaseServer;
-    [Write] string DatabaseName;
+    [Key, Description('The name of the managed metadata service application')] string Name;
+    [Required, Description('The application pool that the service app will use')] string ApplicationPool;
+    [Write, Description('The name of the database server which will host the application')] string DatabaseServer;
+    [Write, Description('The name of the database for the service application')] string DatabaseName;
+    [Write, Description('POWERSHELL 4 ONLY: The account to run this resource as, use PsDscRunAsAccount if using PowerShell 5'), EmbeddedInstance("MSFT_Credential")] String InstallAccount;
 };
```
Comment (BrianFarnhill) [inline Modules/xSharePoint/DSCResources/MSFT_xSPManagedMetadataServiceApp/MSFT_xSPManagedMetaDataServiceApp.schema.mof]: Agreed but will handle that when we go back to close off #148

File: Modules/xSharePoint/DSCResources/MSFT_xSPBCSServiceApp/MSFT_xSPBCSServiceApp.schema.mof [hunk 24]
```diff
@@ -20,10 +20,10 @@ Database names or server name will not be changed if the configuration does not
...
-    [Required] String ApplicationPool;
-    [Write] string DatabaseName;
-    [Write] String DatabaseServer;
-    [Write, EmbeddedInstance("MSFT_Credential")] String InstallAccount;
+    [Key, Description('The name of the BCS service app')] string Name;
+    [Required, Description('The application pool it should run in')] String ApplicationPool;
+    [Write, Description('Name of the database to create for the service app')] string DatabaseName;
+    [Write, Description('Name of the database server to host the database on')] String DatabaseServer;
+    [Write, Description('POWERSHELL 4 ONLY: The account to run this resource as, use PsDscRunAsAccount if using PowerShell 5'), EmbeddedInstance("MSFT_Credential")] String InstallAccount;
 };
```
Comment (BrianFarnhill) [inline Modules/xSharePoint/DSCResources/MSFT_xSPBCSServiceApp/MSFT_xSPBCSServiceApp.schema.mof]: Again for #148

File: Modules/xSharePoint/DSCResources/MSFT_xSPAppManagementServiceApp/MSFT_xSPAppManagementServiceApp.schema.mof (hunk 4 above)
Comment (BrianFarnhill) [inline Modules/xSharePoint/DSCResources/MSFT_xSPAppManagementServiceApp/MSFT_xSPAppManagementServiceApp.schema.mof]: See #148

File: Modules/xSharePoint/DSCResources/MSFT_xSPSearchServiceApp/MSFT_xSPSearchServiceApp.schema.mof (hunk 6 above)
Comment (BrianFarnhill) [inline Modules/xSharePoint/DSCResources/MSFT_xSPSearchServiceApp/MSFT_xSPSearchServiceApp.schema.mof]: See #148

File: Modules/xSharePoint/DSCResources/MSFT_xSPSecureStoreServiceApp/MSFT_xSPSecureStoreServiceApp.schema.mof (hunk 8 above)
Comment (BrianFarnhill) [inline Modules/xSharePoint/DSCResources/MSFT_xSPSecureStoreServiceApp/MSFT_xSPSecureStoreServiceApp.schema.mof]: See #148

File: Modules/xSharePoint/DSCResources/MSFT_xSPSessionStateService/MSFT_xSPSessionStateService.schema.mof (hunk 9 above)
Comment (BrianFarnhill) [inline Modules/xSharePoint/DSCResources/MSFT_xSPSessionStateService/MSFT_xSPSessionStateService.schema.mof]: See #148

File: Modules/xSharePoint/DSCResources/MSFT_xSPStateServiceApp/MSFT_xSPStateServiceApp.schema.mof (hunk 12 above)
Comment (BrianFarnhill) [inline Modules/xSharePoint/DSCResources/MSFT_xSPStateServiceApp/MSFT_xSPStateServiceApp.schema.mof]: See #148

File: Modules/xSharePoint/DSCResources/MSFT_xSPSubscriptionSettingsServiceApp/MSFT_xSPSubscriptionSettingsServiceApp.schema.mof (hunk 13 above)
Comment (BrianFarnhill) [inline Modules/xSharePoint/DSCResources/MSFT_xSPSubscriptionSettingsServiceApp/MSFT_xSPSubscriptionSettingsServiceApp.schema.mof]: See #148

File: Modules/xSharePoint/DSCResources/MSFT_xSPUsageApplication/MSFT_xSPUsageApplication.schema.mof (hunk 14 above)
Comment (BrianFarnhill) [inline Modules/xSharePoint/DSCResources/MSFT_xSPUsageApplication/MSFT_xSPUsageApplication.schema.mof]: See #148

File: Modules/xSharePoint/DSCResources/MSFT_xSPVisioServiceApp/MSFT_xSPVisioServiceApp.schema.mof (hunk 16 above)
Comment (BrianFarnhill) [inline Modules/xSharePoint/DSCResources/MSFT_xSPVisioServiceApp/MSFT_xSPVisioServiceApp.schema.mof]: See #148 - This one actually needs InstallAccount as well. which I'll capture elsewhere

**Pull request merged**